flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/service/AbstractFlinkService.java

```diff
@@ -195,8 +199,25 @@ public void submitApplicationCluster(
         if (requireHaMetadata) {
             validateHaMetadataExists(conf);
         }
+        try {
+            deployApplicationCluster(jobSpec, removeOperatorConfigs(conf));
+        } catch (RuntimeException e) {
+            LOG.warn("Caught Exception " + e.getMessage());
+            if (!isValidRuntimeException(conf, e)) {
+                throw e;
+            }
+        }
```

Comment:
I am very confused by this logic, can we keep the previous logic and just let the exceptions fall through? What's the point of logging them? Deploy errors should result in errors

Reply:
I'll try and explain the problem. If the tls config for the flinkdeployment is different to the operator then on creating (deploying) a FlinkDeployment, a rest client is always returned see https://github.com/apache/flink/blob/c5808b04fdce9ca0b705b6cc7a64666ab6426875/flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesClusterDescriptor.java#L232. This has the tls configuration for the deployment and may not match that of the operator, so will throw an exception even though the rest client is not and will never be used. This exception is then propagated into the CR and looks like its failed. However after a few reconciles it will correct itself. I thought catching this erroneous exception would prevent users thinking they have an issue with their deployment